scala/private/phases/api.bzl

```diff
@@ -59,18 +60,21 @@ def run_phases(ctx, builtin_customizable_phases, fixed_phase):
     # A placeholder for data shared with later phases
     global_provider = {}
     current_provider = struct(**global_provider)
+    rule_providers = []
     for (name, function) in adjusted_phases + [fixed_phase]:
         # Run a phase
         new_provider = function(ctx, current_provider)
 
         # If a phase returns data, append it to global_provider
         # for later phases to access
         if new_provider != None:
+            if (hasattr(new_provider, "rule_providers")):
+                rule_providers.extend(new_provider.rule_providers)
```

Comment:
`rule_providers = []` and `new_provider.rule_providers` have the same name. Could we have at least a little bit of difference? Like maybe `new_provider.rule_provider`.

Reply:
I actually thought about changing the strict field name to be `inter_rule_providers` or `external_providers` and then the next phase will be to have phases also export `internal_providers` between phases (I think phases_providers is overloaded). I’ll also change the temp variable to something like what you suggested

Reply:
Thanks, it looks good.